repository_service_tuf/cli/admin/import_targets.py

```diff
@@ -0,0 +1,202 @@
+from typing import Any, Dict, List
+
+from rich.console import Console
+from sqlalchemy import Connection, MetaData, Table, create_engine
+from sqlalchemy.exc import IntegrityError
+
+from repository_service_tuf.cli import click
+from repository_service_tuf.cli.admin import admin
+from repository_service_tuf.helpers.api_client import (
+    URL,
+    Methods,
+    get_headers,
+    is_logged,
+    publish_targets,
+    request_server,
+    task_status,
+)
+from repository_service_tuf.helpers.tuf import (
+    Delegations,
+    Metadata,
+    SuccinctRoles,
+    Targets,
+)
+
+console = Console()
+
+
+def _check_csv_files(csv_files: List[str]):
+    not_found_csv_files: List[str] = []
+    for csv_file in csv_files:
+        if not os.path.isfile(csv_file):
+            not_found_csv_files.append(csv_file)
+
+    if len(not_found_csv_files) > 0:
+        raise click.ClickException(
+            f"CSV file(s) not found: {(', ').join(not_found_csv_files)}"
+        )
+
+
+def _parse_csv_data(
+    csv_file: str, succinct_roles: SuccinctRoles
+) -> List[Dict[str, Any]]:
+    rstuf_db_data: List[Dict[str, Any]] = []
+    with open(csv_file, "r") as f:
+        for line in f:
+            rstuf_db_data.append(
+                {
+                    "path": line.split(";")[0],
+                    "info": {
+                        "length": int(line.split(";")[1]),
+                        "hashes": {line.split(";")[2]: line.split(";")[3]},
+                    },
+                    "rolename": succinct_roles.get_role_for_target(
+                        line.split(";")[0]
+                    ),
+                    "published": False,
+                    "action": "ADD",
+                    "last_update": datetime.now(),
+                }
+            )
+
+    return rstuf_db_data
+
+
+def _import_csv_to_rstuf(
+    db_client: Connection,
+    rstuf_table: Table,
+    csv_files: List[str],
+    succinct_roles: SuccinctRoles,
+) -> None:
+    for csv_file in csv_files:
+        console.print(f"Import status: Loading data from {csv_file}")
+        rstuf_db_data = _parse_csv_data(csv_file, succinct_roles)
+        console.print(f"Import status: Importing {csv_file} data")
+        try:
+            db_client.execute(rstuf_table.insert(), rstuf_db_data)
+        except IntegrityError:
+            raise click.ClickException(
+                "Import status: ABORTED due duplicated targets. "
+                "CSV files must to have unique targets (path). "
+                "No data added to RSTUF DB."
+            )
+        console.print(f"Import status: {csv_file} imported")
+
+
+def _get_succinct_roles(metadata_url: str) -> SuccinctRoles:
+    response = request_server(metadata_url, "1.bin.json", Methods.get)
+    if response.status_code == 404:
+        raise click.ClickException("RSTUF Metadata Targets not found.")
+
+    json_data = json.loads(response.text)
+    targets: Metadata[Targets] = Metadata.from_dict(json_data)
```

Comment:
> I spotted it here, but if you validate it I can open a bug. (BTW quick fix: `data = response.json().get("data", {})`)

That's is a good spot, I'm implementing the quick fix.

> On the other hand, we are always validating a token with the same token in the headers, so I assume if the token is invalid it won't be because it can't be decoded... so it might be fine this way. I will not delete it, but I believe you can ignore my comment.

Maybe we can have a specific issue for this part to implement or remove this check.
Can you file the issue giving the background?

Reply:
As we discussed, it gets the readability confused. So we decided to keep it as is.